### PR TITLE
dev/Unity2018 - Updating projects to Unity2018 assemblies

### PR DESCRIPTION
**Pull request by PrimeSonic**
- DLLs taken from the Subnautica Experimental branch
- Identical to DLLs in Below Zero

### CODE REVIEW

File: QModManager/API/SMLHelper/Utility/ImageUtils.cs [hunk 1]
```diff
@@ -28,13 +28,16 @@ public static Texture2D LoadTextureFromFile(string filePathToImage, TextureForma
...
                 {
+                    texture2D.LoadRawTextureData(imageBytes);
+
                     return texture2D;
                 }
-                else
+                catch (UnityException uex)
                 {
-                    Logger.Error("Error on LoadTextureFromFile call. Texture cannot be loaded: " + filePathToImage);
+                    Logger.Error("Error on LoadTextureFromFile call. Texture cannot be loaded: " + filePathToImage + uex);
```
Comment (AlexejheroYTB) [inline QModManager/API/SMLHelper/Utility/ImageUtils.cs]:
```csharp
Logger.Error(...);
Logger.Exception(uex);
```

File: QModManager/QModManager.csproj [hunk 2]
```diff
@@ -104,12 +104,17 @@
     <Reference Include="System.Data" />
     <Reference Include="System.Xml" />
     <Reference Include="UnityEngine">
+      <SpecificVersion>False</SpecificVersion>
       <HintPath>..\Dependencies\UnityEngine.dll</HintPath>
```
Comment (AlexejheroYTB) [inline QModManager/QModManager.csproj]: Do we need both `UnityEngine.dll` and the other unity files referenced?

File: QModManager/QModManager.csproj [hunk 3]
```diff
@@ -104,12 +104,17 @@
...
-      <Private>False</Private>
+    </Reference>
+    <Reference Include="UnityEngine.CoreModule">
+      <HintPath>..\Dependencies\UnityEngine.CoreModule.dll</HintPath>
+    </Reference>
+    <Reference Include="UnityEngine.PhysicsModule">
+      <HintPath>..\Dependencies\UnityEngine.PhysicsModule.dll</HintPath>
```
Comment (AlexejheroYTB) [inline QModManager/QModManager.csproj]: `UnityEngine.dll` split into 50 dlls, I'm assuming that we are only referencing the ones needed?

Comment (PrimeSonic) [reply QModManager/QModManager.csproj]: Correct

File: QModManager/QModManager.csproj (hunk 2 above)
Comment (PrimeSonic) [reply QModManager/QModManager.csproj]: Actually yes. `Monobehavior` is still defined within the new "UnityEngine.dll".  
Project won't compile without it.

File: Examples/Examples.csproj [hunk 4]
```diff
@@ -43,8 +43,8 @@
     <Reference Include="System.Data.DataSetExtensions" />
     <Reference Include="System.Data" />
     <Reference Include="System.Xml" />
-    <Reference Include="UnityEngine">
-      <HintPath>..\Dependencies\UnityEngine.dll</HintPath>
```
Comment (PrimeSonic) [inline Examples/Examples.csproj]: Changes need to be reflected in /Dependencies/README.md

**Pull request merged**